yt_dlp/networking/common.py

```diff
@@ -517,6 +517,7 @@ def __init__(
             self.reason = reason or HTTPStatus(status).phrase
         except ValueError:
             self.reason = None
+        self.extensions = {}
```

Comment:
I would also expose this as a parameter for Response, same as Request

+ add in docstring